rosbag2_storage_default_plugins/src/rosbag2_storage_default_plugins/sqlite/sqlite_storage.cpp

```diff
@@ -56,6 +69,37 @@ bool is_read_write(const rosbag2_storage::storage_interfaces::IOFlag io_flag)
   return io_flag == rosbag2_storage::storage_interfaces::IOFlag::READ_WRITE;
 }
 
+inline std::vector<std::string> parse_pragmas(
+  const std::string & storage_config_uri, const rosbag2_storage::storage_interfaces::IOFlag io_flag)
+{
+  std::vector<std::string> pragmas;
+  if (!storage_config_uri.empty()) {
+    try {
+      auto key =
+        io_flag == rosbag2_storage::storage_interfaces::IOFlag::READ_ONLY ? "read" : "write";
+      YAML::Node yaml_file = YAML::LoadFile(storage_config_uri);
+      pragmas = yaml_file[key]["pragmas"].as<std::vector<std::string>>();
+    } catch (const YAML::Exception & ex) {
+      throw std::runtime_error(std::string("Exception on parsing info file: ") + ex.what());
+    }
+    // poor developer's sqlinjection prevention ;-)
+    std::string invalid_characters = {"';\""};
+    auto throw_on_invalid_character = [](const auto & pragmas, const auto & invalid_characters) {
+        for (const auto & pragma_string : pragmas) {
+          auto pos = pragma_string.find_first_of(invalid_characters);
+          if (pos != std::string::npos) {
+            throw std::runtime_error(
```

Comment:
it might be useful to print the list of disallowed characters along with the whole pragma string, so it's more clear what they did wrong.

Reply:
I added this check in the PR to this branch #545